runtime/runtime/src/metrics.rs

```diff
@@ -558,3 +616,54 @@ impl ApplyMetrics {
             .observe(self.accumulated_compute as f64 / TERA);
     }
 }
+
+pub fn report_congestion_metrics(receipt_sink: &ReceiptSink, sender_shard_id: ShardId) {
+    match receipt_sink {
+        ReceiptSink::V1(_) => {
+            // no metrics to report
+        }
+        ReceiptSink::V2(inner) => {
+            let sender_shard_label = sender_shard_id.to_string();
+            report_congestion_indicators(&inner.congestion_info, &sender_shard_label);
+            report_outgoing_buffers(inner, sender_shard_label);
+        }
+    }
+}
+
+/// Report key congestion indicator levels of a shard.
+fn report_congestion_indicators(congestion_info: &CongestionInfo, shard_label: &str) {
+    let congestion_level = congestion_info.congestion_level();
+    CONGESTION_LEVEL.with_label_values(&[shard_label]).set(congestion_level);
+
+    let CongestionInfo::V1(inner) = congestion_info;
+    CONGESTION_RECEIPT_BYTES
+        .with_label_values(&[shard_label])
+        .set(inner.receipt_bytes.try_into().unwrap_or(i64::MAX));
+    CONGESTION_INCOMING_GAS
+        .with_label_values(&[shard_label])
+        .set(inner.delayed_receipts_gas.try_into().unwrap_or(i64::MAX));
+    CONGESTION_OUTGOING_GAS
+        .with_label_values(&[shard_label])
+        .set(inner.buffered_receipts_gas.try_into().unwrap_or(i64::MAX));
```

Comment:
For the sake of understanding - why do you need try_into? Is that because you're casting from u128 to i64?

Reply:
u128 or u64, both have number higher than i64 can represent. But I think it's fine to cap that at i64::MAX for metrics.